.github/workflows/build.yml

```diff
@@ -31,9 +31,42 @@ jobs:
         ports:
           - 6379:6379
     steps:
+      - name: Checkout master
+        uses: actions/checkout@v3
+        with:
+          ref: master
+
       - name: Checkout
         uses: actions/checkout@v3
 
+      - name: Check migration conflicts
+        run: |
+          NEW_MIGRATIONS_COUNT=$(ls packages/server/postgres/migrations | wc -l)
+          MAX_NEW_MIGRATION=$(ls packages/server/postgres/migrations | tail -n 1)
+
+          git checkout master
+          OLD_MIGRATIONS_COUNT=$(ls packages/server/postgres/migrations | wc -l)
+          MAX_OLD_MIGRATION=$(ls packages/server/postgres/migrations | tail -n 1)
+
+          git checkout ${{ github.ref_name }}
+
+          if [ $OLD_MIGRATIONS_COUNT == $NEW_MIGRATIONS_COUNT ]; then
+            echo No new migrations found
+            exit 0
+          elif [[ 'OLD_MIGRATIONS_COUNT + 1' -lt 'NEW_MIGRATIONS_COUNT' ]]; then
+            # Necessary since we only check the newest migration, not each one
+            echo Please only PR 1 Migration at a time
+            exit 1
+          fi
```

Comment:
+1 I'm not a fan of this restriction. If you know the number of migrations, you can do
```
FIRST_NEW_MIGRATION=$(ls packages/server/postgres/migrations | tail -n $NEW_MIGRATION_COUNT | head -n1)
```